src/Constraint/Constraint.php

```diff
@@ -171,6 +171,73 @@ public function versionCompare($a, $b, $operator, $compareBranches = false)
         return version_compare($a, $b, $operator);
     }
 
+    public function compile($otherOperator) {
+        $otherOperator = self::$transOpStr[$otherOperator];
+
+        $isBranch = 'dev-' === substr($this->version, 0, 4);
+
+        if (self::OP_NE === $this->operator) {
+            if ($isBranch) {
+                if ($otherOperator === self::OP_EQ) {
+                    return 'false';
+                }
+
+                return 'true';
+            }
+
+            if ($otherOperator === self::OP_EQ) {
+                return sprintf('!$b && version_compare($v, %s, \'!=\')', \var_export($this->version, true));
+            }
+
+            return 'true';
+        } elseif (self::OP_EQ === $this->operator) {
+            if ($isBranch) {
+                if (in_array($otherOperator, array(self::OP_LT, self::OP_GT, self::OP_NE))) {
+                    return 'false';
+                }
+
+                return sprintf('$b && $v === %s', \var_export($this->version, true));
+            }
+
+            if ($otherOperator === self::OP_NE) {
+                return sprintf('(!$b && version_compare($v, %s, \'!=\'))', \var_export($this->version, true));
+            } elseif (in_array($otherOperator, array(self::OP_LT, self::OP_GT))) {
+                return 'false';
+            }
+
+            return sprintf('version_compare($v, %s, \'==\')', \var_export($this->version, true));
+        }
+
+        if (in_array($this->operator, array(self::OP_LT, self::OP_LE))) {
+            if (in_array($otherOperator, array(self::OP_LT, self::OP_LE))) {
+                return 'true';
+            }
+        } elseif (in_array($otherOperator, array(self::OP_GT, self::OP_GE))) {
+            return 'true';
+        }
+
+        if ($otherOperator === self::OP_NE) {
+            return 'true';
+        }
+
+        if ($isBranch) {
+            return 'false';
+        }
+
+        $codeComparison = sprintf('version_compare($v, %s, \'%s\')', \var_export($this->version, true), self::$transOpInt[$this->operator]);
```

Comment:
Should the compiled code also be fully qualified?

Reply:
indeed